src/core/cfl.cpp

```diff
@@ -27,8 +26,7 @@ void setup(nrs_t* nrs)
     free(dH);
   }
 
-  tmp = (dfloat*) calloc(nrs->Nblock, sizeof(dfloat));
-  o_tmp = mesh->device.malloc(nrs->Nblock * sizeof(dfloat), tmp);
+  tmp = (dfloat*) calloc(mesh->Nelements, sizeof(dfloat));
```

Comment:
Please use the existing scratch space instead of allocating yet another tmp array.